docs/contributing/testing.md

```diff
@@ -0,0 +1,74 @@
+# Testing
+
+When making a contribution to opsdroid, it\'s important that tests are
+written to validate that the functionality works, and continues to work
+in the future.
+
+opsdroid is currently slowly migrating from
+[unittest](https://docs.python.org/3.8/library/unittest.html) to
+[pytest](https://docs.pytest.org) as it\'s testing framework. This work
+is tracked in issue
+[\#1502](https://github.com/opsdroid/opsdroid/issues/1502). As a part of
+this migration tests can be found in one of two places, old unittest
+tests are found in the `tests/` directory in the root of the repository
+and new pytest tests are found in a `tests/` directory in the same
+subfolder as the code, i.e `opsdroid/tests/`.
```

Comment:
There appear to be some errant `\`s going on in here.

Reply:
Also why so many line breaks?

Reply:
oh I just let pandoc loose on it. :confused:

Reply:
fixed.